risc0/zkp/src/core/sha_cpu.rs

```diff
@@ -35,31 +34,40 @@ fn set_word(buf: &mut [u8], idx: usize, word: u32) {
 
 impl Impl {
     /// Compute the hash of a slice of plain-old-data using the
-    /// specified offset and stride.  'size' specifies the number of
+    /// specified offset and stride. 'size' specifies the number of
     /// elements to hash.
     pub fn hash_pod_stride<T: bytemuck::Pod>(
-        &self,
         pods: &[T],
         offset: usize,
         size: usize,
         stride: usize,
     ) -> Box<Digest> {
-        let mut state = *SHA256_INIT.get();
+        // Because bytes in our digests are in big-endian order, regardless of host
+        // architecture, and the `sha2::compress256` function takes words in native
+        // byte order, the bytes of the IV need to be flipped on little-endian
+        // machines (e.g. x86).
+        let mut state: [u32; DIGEST_WORDS] = SHA256_INIT.into();
         for word in state.iter_mut() {
             *word = word.to_be();
         }
 
         let mut block: GenericArray<u8, U64> = GenericArray::default();
 
+        // Construct an Iterator<u8> with the given, offset, stride, and size in items.
         let mut bytes = pods
             .iter()
             .skip(offset)
             .step_by(stride)
             .take(size)
             .flat_map(|pod| bytemuck::cast_slice(slice::from_ref(pod)) as &[u8])
-            .cloned()
+            .copied()
             .fuse();
 
+        // TODO(victor): It looks like this function _does not_ implement canonical
```

Comment:
This call should only occur inside Seal verification of data structures whose sizes are determined by the protocol (i.e. collisions of different sized data should never be relevant).  That said, the cost to doing standard padding in this case is minimal, so perhaps it's a good idea.